packages/aws-cdk/lib/init-templates/app/csharp/add-project.hook.ts

```diff
@@ -1,33 +1,9 @@
-import * as child_process from 'child_process';
 import * as path from 'path';
 import { InvokeHook } from '../../../init';
+import { shell } from '../util/os';
 
 export const invoke: InvokeHook = async (targetDirectory: string) => {
   const slnPath = path.join(targetDirectory, 'src', '%name.PascalCased%.sln');
   const csprojPath = path.join(targetDirectory, 'src', '%name.PascalCased%', '%name.PascalCased%.csproj');
-
-  const child = child_process.spawn('dotnet', ['sln', slnPath, 'add', csprojPath], {
-    // Need this for Windows where we want .cmd and .bat to be found as well.
-    shell: true,
-    stdio: ['ignore', 'pipe', 'inherit'],
-  });
-
-  await new Promise<string>((resolve, reject) => {
-    const stdout = new Array<any>();
-
-    child.stdout.on('data', chunk => {
-      process.stdout.write(chunk);
-      stdout.push(chunk);
-    });
-
-    child.once('error', reject);
-
-    child.once('exit', code => {
-      if (code === 0) {
-        resolve(Buffer.concat(stdout).toString('utf-8'));
-      } else {
-        reject(new Error(`Could not add project %name.PascalCased%.csproj to solution %name.PascalCased%.sln. Error code: ${code}`));
-      }
-    });
-  });
-};
+  await shell(['dotnet', 'sln', slnPath, 'add', csprojPath], { errorMessage: 'Could not add project %name.PascalCased%.csproj to solution %name.PascalCased%.sln.' });
```

Comment:
What's the advantage of adding the `errorMessage` feature over just adding a `try/catch` here?

Seems more orthogonal to leave it to consumer to decide how to handle errors.